huaweicloud/data_source_huaweicloud_gaussdb_mysql_instances_test.go

```diff
@@ -0,0 +1,84 @@
+package huaweicloud
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccGaussdbMysqlInstancesDataSource_basic(t *testing.T) {
+	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGaussdbMysqlInstancesDataSource_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGaussdbMysqlInstancesDataSourceID("data.huaweicloud_gaussdb_mysql_instances.test"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGaussdbMysqlInstancesDataSourceID(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Can't find GaussDB mysql instance data source: %s ", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("GaussDB mysql instances data source ID not set ")
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckGaussdbMysqlInstancesDataLen(n string) resource.TestCheckFunc {
```

Comment:
seems this function is never called.

BTW, this check funcction is not recommended, you can try `resource.TestCheckResourceAttr("data.huaweicloud_gaussdb_mysql_instances.test", "instances.#", "1"),`

Reply:
fixed

Reply:
new test result:
```bash
make testacc TEST='./huaweicloud' TESTARGS='-run TestAccGaussdbMysqlInstancesDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccGaussdbMysqlInstancesDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccGaussdbMysqlInstancesDataSource_basic
=== PAUSE TestAccGaussdbMysqlInstancesDataSource_basic
=== CONT  TestAccGaussdbMysqlInstancesDataSource_basic
--- PASS: TestAccGaussdbMysqlInstancesDataSource_basic (813.94s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       813.983s
```